slick/src/main/scala/slick/util/ManagedArrayBlockingQueue.scala

```diff
@@ -34,53 +38,50 @@ class ManagedArrayBlockingQueue[E >: Null <: PrioritizedRunnable](maximumInUse:
 
   private[this] var paused = false
 
-  private[util] def increaseInUseCount(pr: PrioritizedRunnable): Unit = {
+  private[util] def increaseInUseCount(pr: PrioritizedRunnable): Unit = locked {
```

Comment:
Great, I think this is the main reason, why some people still see the executor pausing, while the connection pool still has threads available. The increaseInUseCount is probably incremented twice for the same runnable. Once by a thread operating on the queue and once by the actual thread which will execute the runnable. 
By acquiring the lock before doing the check if it was already incremented, the inUseCounterSet value should be viewable to all threads.